test/user.integration.js

```diff
@@ -75,7 +69,12 @@ describe('users - integration', function() {
 
           expect(res.body.title).to.be.eql('T1');
           expect(res.body.content).to.be.eql('C1');
-          expect(res.body.userId).to.be.eql(userId);
+          // HACK(bajtos) other tests are somehow screwing up the state of
+          // models and/or datasource, and as a result, body.userId is string
+          // while userId is a number.
+          // This cannot be reproduced when the tests in this file are run
+          // in isolation.
```

Comment:
Here is the offending line:

```js
User.hasMany(Post);
```

https://github.com/strongloop/loopback/blob/8f7e032a01bdde54b195385dc5eff37dc4524739/test/model.test.js#L105

Looks like when the user-integration-app defines "User hasMany Post" relation, the Post model defined here is looked up instead of the "post" model defined by that app.

/cc @raymondfeng @ritch any idea on why is that happening? I'll continue my investigation, I'm just wondering if you perhaps can save me some time.